ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,168 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Tests to verify that the ipa-healthcheck scenarios
+"""
+
+from __future__ import absolute_import
+
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+import json
+
+HEALTHCHECK_LOG = "/var/log/ipa/healthcheck/healthcheck.log"
+HEALTHCHECK_SYSTEMD_FILE = (
+    "/etc/systemd/system/multi-user.target.wants/ipa-healthcheck.timer"
+)
+HEALTHCHECK_LOGROTATE_CONF = "/etc/logrotate.d/ipahealthcheck"
+HEALTHCHECK_LOG_DIR = "/var/log/ipa/healthcheck"
+
+
+def format_output(outputfile):
+    """
+    Helper function which parses the output.json file which is
+    created when ipa-healthcheck command is executed
+    """
+    with open(outputfile, "r") as f:
+        data = json.load(f)
+    return data
+
+
+sources = [
+    "ipahealthcheck.dogtag.ca",
+    "ipahealthcheck.ds.replication",
+    "ipahealthcheck.dogtag.ca",
+    "ipahealthcheck.ipa.certs",
+    "ipahealthcheck.ipa.dna",
+    "ipahealthcheck.ipa.idns",
+    "ipahealthcheck.ipa.files",
+    "ipahealthcheck.ipa.host",
+    "ipahealthcheck.ipa.roles",
+    "ipahealthcheck.ipa.topology",
+    "ipahealthcheck.ipa.trust",
+    "ipahealthcheck.meta.services",
+]
+
+ipa_cert_checks = [
+    "IPACertmongerExpirationCheck",
+    "IPACertfileExpirationCheck",
+    "IPACertTracking",
+    "IPACertNSSTrust",
+    "IPANSSChainValidation",
+    "IPAOpenSSLChainValidation",
+    "IPARAAgent",
+    "IPACertRevocation",
+    "IPACertmongerCA",
+    "IPACAChainExpirationCheck",
+]
+
+ipatrust_checks = [
+    "IPATrustAgentCheck",
+    "IPATrustDomainsCheck",
+    "IPADomainCheck",
+    "IPATrustCatalogCheck",
+    "IPAsidgenpluginCheck",
+    "IPATrustAgentMemberCheck",
+    "IPATrustControllerPrincipalCheck",
+    "IPATrustControllerServiceCheck",
+    "IPATrustControllerConfCheck",
+    "IPATrustControllerGroupSIDCheck",
+    "IPATrustPackageCheck",
+]
+
+metaservices_checks = [
+    "certmonger",
+    "dirsrv",
+    "gssproxy",
+    "httpd",
+    "ipa_custodia",
+    "ipa_dnskeysyncd",
+    "ipa_otpd",
+    "kadmin",
+    "krb5kdc",
+    "named",
+    "pki_tomcatd",
+    "sssd",
+]
+
+ipafiles_checks = ["IPAFileNSSDBCheck", "IPAFileCheck", "TomcatFileCheck"]
+dogtag_checks = ["DogtagCertsConfigCheck", "DogtagCertsConnectivityCheck"]
+iparoles_checks = ["IPACRLManagerCheck", "IPARenewalMasterCheck"]
+replication_checks = ["ReplicationConflictCheck"]
+ruv_checks = ["RUVCheck"]
+dna_checks = ["IPADNARangeCheck"]
+idns_checks = ["IPADNSSystemRecordsCheck"]
+ipahost_checks = ["IPAHostKeytab"]
+ipatopology_checks = ["IPATopologyDomainCheck"]
+filesystem_checks = ["FileSystemSpaceCheck"]
+metacore_checks = ["MetaCheck"]
+
+
+class TestIpaHealthCheck(IntegrationTest):
+    """
+    Check that ipa healthcheck tool return WARNING if no range is set.
+    https://github.com/freeipa/freeipa-healthcheck/issues/60
+    """
```

Comment:
The comment doesn't match the content. This is just verifying that the default set of sources and checks are available.